clusterDeployer.py

```diff
@@ -119,15 +119,19 @@ def teardown(self) -> None:
         removed_macs = []
         names = [x.name for x in self._cc.all_vms()]
         ips = [x.ip for x in self._cc.all_vms()]
-        for e in q[-1][0][1:]:
-            if e.attrib["name"] in names or e.attrib["ip"] in ips:
-                mac = e.attrib["mac"]
-                name = e.attrib["name"]
-                ip = e.attrib["ip"]
-                pre = "virsh net-update default delete ip-dhcp-host"
-                cmd = f"{pre} \"<host mac='{mac}' name='{name}' ip='{ip}'/>\" --live --config"
-                logger.info(self._local_host.hostconn.run(cmd))
-                removed_macs.append(mac)
+        ip_tree = q.find('ip')
+        if ip_tree:
+            dhcp = ip_tree.find('dhcp')
+            if dhcp:
+                for e in dhcp:
+                    if e.get('name') in names or e.get('ip') in ips:
```

Comment:
This is nested 4 levels deep. Can you refactor it to avoid deep nesting?

Reply:
Ack